docs/hacking-an-agent.md

```diff
@@ -1,99 +1,222 @@
+wallet = Wallet({'default': 'my_key.txt'})
+stub_connection = StubConnection(input_file_path=input_filename, output_file_path=output_filename)
+ledger_apis = LedgerApis({})
+resources = Resources('')
+
+# Create our AEA
+my_agent = AEA("my_agent", [stub_connection], wallet, ledger_apis, resources)
 ```
 
-Create a wallet object with a private key.
+Create the default protocol and add it to the agent
 ``` python
-wallet = Wallet({'default': 'my_key.txt'})
+# Add the default protocol
+default_protocol_configuration = ProtocolConfig.from_json(
+    yaml.safe_load(open(os.path.join(AEA_DIR, "protocols", "default", "protocol.yaml"))))
+default_protocol = Protocol("default", DefaultSerializer(), default_protocol_configuration)
+resources.protocol_registry.register(("default", None), default_protocol)
 ```
 
-Create a `Connection`.
+Create the error skill (needed by all agents) and the echo skill which will bounce our messages back to us
 ``` python
-stub_connection = StubConnection(input_file_path='input.txt', output_file_path='output.txt')
+# Add the error skill and the echo skill
+echo_skill = Skill.from_dir(os.path.join(root_dir, "packages", "skills", "echo"), my_agent.context)
+resources.add_skill(echo_skill)
+error_skill = Skill.from_dir(os.path.join(AEA_DIR, "skills", "error"), my_agent.context)
+resources.add_skill(error_skill)
 ```
 
-For ledger APIs, we simply feed the agent an empty dictionary (meaning we do not require any). 
+## Start the agent
+We run the agent from a different thread so that we can still use the main thread to pass it messages.
 ``` python
-ledger_apis = LedgerApis({})
+# Set the agent running in a different thread
+t = Thread(target=my_agent.start)
+t.start()
+
+# Wait for everything to start up
+time.sleep(4)
 ```
 
-Create the resources pointing to the working directory.
+## Send and receive an envelope
+We use the input and output text files to send an envelope to our agent and recieve a response (from the echo skill)
 ``` python
-resources = Resources('')
+# Create a message inside an envelope and get the stub connection to pass it on to the echo skill
+message_text = 'my_agent,other_agent,default,{"type": "bytes", "content": "aGVsbG8="}'
+with open(input_filename, 'w') as f:
+    f.write(message_text)
+    print("input message: " + message_text)
+
+# Wait for the envelope to get processed
+time.sleep(4)
+
+# Read the output envelope generated by the echo skill
+with open(output_filename, 'r') as f:
+    print("output message: " + f.readline())
 ```
 
-Now we have everything we need for initialisation.
+## Shutdown
+Finally stop our agent and wait for it to finish
 ``` python
-my_agent = AEA("my_agent", stub_connection, wallet, ledger_apis, resources)
+# Shut down the agent
+my_agent.stop()
+t.join()
 ```
 
-## Add skills and protocols
+## Running the agent
+If you now run this pythong script file, you should see this output:
 
-We can add the echo skill as follows...
+    No protocol found.
```

Comment:
Hmm..what causes these log messages? I see it is caused by line 150 in `aea.registries.base`. We should not raise this warning in this case. @marcofavorito should we just change the logger level or come up with a more elaborate solution?